### PR TITLE
ADM-928:[frontend]feat: add metrics failed date range viewer icon

### PR DESCRIPTION
**Pull request by weiraneve**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

**Review by PengxiWPix — APPROVED** [review]
LGTM

File: frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx [hunk 1]
```diff
@@ -42,4 +52,19 @@ describe('DateRangeViewer', () => {
     expect(getByText(/2024\/03\/19/)).toBeInTheDocument();
     expect(getByText(/2024\/03\/21/)).toBeInTheDocument();
   });
+
+  it('should show priority high icon when click expand button and step number is 1', async () => {
+    store.dispatch(nextStep());
+    store.dispatch(updateFailedTimeRange(['1706716800000']));
```
Comment (mrcuriosity-tw) [inline frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx:58]: const a name to express what does `1706716800000` mean.

File: frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx [hunk 2]
```diff
@@ -42,4 +52,19 @@ describe('DateRangeViewer', () => {
...
+    const { getByLabelText } = setup(mockDateRanges);
+    await userEvent.click(getByLabelText('expandMore'));
+    expect(screen.getByTestId('PriorityHighIcon')).toBeInTheDocument();
+  });
+
+  it('should not show priority high icon when click expand button and step number is 0', async () => {
+    store.dispatch(backStep());
```
Comment (mrcuriosity-tw) [inline frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx:65]: Among test cases, they should be as stateless as possible, hence, better to reset the store before/after each, and new backStep here.
Imagine that someone accidently add a case and have some dependency with step, it probably fails.

**Review by TingyuDong — APPROVED** [review]
LGTM!

File: frontend/src/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection/index.tsx [hunk 3]
```diff
@@ -156,7 +156,7 @@ export const PipelineMetricSelection = ({
     if (!isLoading) {
       popup();
     }
-  }, [stepFailedStatus, dispatch, isLoading]);
+  }, [stepFailedStatus, dispatch, isLoading, failedTimeRange]);
```
Comment (mrcuriosity-tw) [inline frontend/src/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection/index.tsx:159]: depend on `failedTimeRange` but I didn't find anywhere used `failedTimeRange`?

**Review by mrcuriosity-tw — APPROVED** [review]

File: frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx [hunk 4]
```diff
@@ -42,4 +60,19 @@ describe('DateRangeViewer', () => {
     expect(getByText(/2024\/03\/19/)).toBeInTheDocument();
     expect(getByText(/2024\/03\/21/)).toBeInTheDocument();
   });
+
+  it('should show priority high icon when click expand button and step number is 1', async () => {
+    const failedTimeRanges = ['1706716800000'];
```
Comment (guzhongren) [inline frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx:65]: @weiraneve change `1706716800000` as a readable varible, covert from a date object

File: frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx [hunk 5]
```diff
@@ -42,4 +60,19 @@ describe('DateRangeViewer', () => {
...
+    const { getByLabelText } = setup(mockDateRanges);
+    await userEvent.click(getByLabelText('expandMore'));
+    expect(screen.getByTestId('PriorityHighIcon')).toBeInTheDocument();
+  });
+
+  it('should not show priority high icon when click expand button and step number is 0', async () => {
+    const { getByLabelText } = setup(mockDateRanges);
+    await userEvent.click(getByLabelText('expandMore'));
+    expect(screen.queryByTestId('PriorityHighIcon')).not.toBeInTheDocument();
+  });
```
Comment (guzhongren) [inline frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx:77]: new line between `given when then`

File: frontend/src/hooks/useGetMetricsStepsEffect.ts [hunk 6]
```diff
@@ -48,6 +49,14 @@ export const useGetMetricsStepsEffect = (): useGetMetricsStepsEffectInterface =>
     );
     const hasRejected = allStepsRes.some((stepInfo) => stepInfo.status === 'rejected');
     const hasFulfilled = allStepsRes.some((stepInfo) => stepInfo.status === 'fulfilled');
+    const rejectedIndices = allStepsRes.reduce((indices: number[], stepInfo, index) => {
+      if (stepInfo.status === 'rejected') {
```
Comment (guzhongren) [inline frontend/src/hooks/useGetMetricsStepsEffect.ts:53]: refactor `rejected` as a fixture, also refactor it in the `useGenerateReportEffect.ts`

File: frontend/src/hooks/useGetBoardInfo.ts [hunk 7]
```diff
@@ -64,9 +67,11 @@ const errorStatusMap = (status: METRICS_DATA_FAIL_STATUS) => {
 };
 
 export const useGetBoardInfoEffect = (): useGetBoardInfoInterface => {
+  const dispatch = useAppDispatch();
   const [isLoading, setIsLoading] = useState(false);
   const [errorMessage, setErrorMessage] = useState({});
   const [boardInfoFailedStatus, setBoardInfoFailedStatus] = useState(METRICS_DATA_FAIL_STATUS.NOT_FAILED);
+  const localFailedTimeRange: string[] = [];
```
Comment (guzhongren) [inline frontend/src/hooks/useGetBoardInfo.ts:74]: rename `localFailedTimeRange` to `localFailedTimeRangeList`

File: frontend/src/context/stepper/StepperSlice.tsx [hunk 8]
```diff
@@ -23,23 +25,34 @@ export const stepperSlice = createSlice({
...
+      }
       state.shouldMetricsLoad = true;
       state.stepNumber += 1;
     },
     backStep: (state) => {
       state.shouldMetricsLoad = false;
       state.stepNumber = state.stepNumber === ZERO ? ZERO : state.stepNumber - 1;
     },
+    updateShouldMetricsLoad: (state, action) => {
+      state.shouldMetricsLoad = action.payload;
```
Comment (guzhongren) [inline frontend/src/context/stepper/StepperSlice.tsx:39]: shouldMetricsLoad -> shouldMetricsLoaded

Comment (guzhongren) [reply frontend/src/context/stepper/StepperSlice.tsx:39]: updateShouldMetricsLoad -> updateShouldMetricsLoaded

File: frontend/src/context/stepper/StepperSlice.tsx [hunk 9]
```diff
@@ -6,12 +6,14 @@ export interface StepState {
...
   timeStamp: number;
   shouldMetricsLoad: boolean;
+  failedTimeRange: string[];
 }
 
 const initialState: StepState = {
   stepNumber: 0,
   timeStamp: 0,
   shouldMetricsLoad: true,
+  failedTimeRange: [],
```
Comment (guzhongren) [inline frontend/src/context/stepper/StepperSlice.tsx:16]: failedTimeRange -> failedTimeRangeList

File: frontend/package.json [hunk 10]
```diff
@@ -12,8 +12,8 @@
     "lint": "eslint -c .eslintrc.json ./ && prettier --check ./",
     "fix": "eslint -c .eslintrc.json --fix && npx prettier --write . --ignore-unknown",
     "audit": "npx audit-ci@^6 --config ./audit-ci.jsonc",
-    "test": "jest",
-    "test:watch": "jest --watchAll",
+    "test": "jest --env=jsdom",
```
Comment (guzhongren) [inline frontend/package.json:15]: could u put env into `jest.config.json` via `https://jestjs.io/docs/configuration#testenvironment-string`

**Review by guzhongren — COMMENTED** [review]
Change with the comments
@weiraneve

**Pull request merged**